packages/@react-spectrum/dropzone/src/DropZone.tsx

```diff
@@ -35,12 +37,17 @@ function DropZone(props: SpectrumDropZoneProps, ref: DOMRef<HTMLDivElement>) {
   let domRef = useDOMRef(ref);
   let messageId = useId();
   let stringFormatter = useLocalizedStringFormatter(intlMessages);
+  let isVirtualDrag = isVirtualDragging(); 
+  let dragSession = DragManager.useDragSession();
+
+  useEffect(() => {
+  }, [dragSession]);
```

Comment:
Moved the `isVirtualDrag` to RSP DropZone so that we only pass the `aria-labelledby` of the filled message when the dropzone is in both a filled and virtual drag state. 

The reason for including `useDragSession` is because most of the re-rendering is happening in RAC DropZone. So when you call `isVirtualDragging` by itself in RSP DropZone, the value isn’t getting updated. `useDragSession` basically allows us to subscribe to when `beginDragging` is called so the a re-render in RSP DropZone can happen and `isVirtualDragging`’s value can be updated.

Reply:
why is the useEffect empty?

Reply:
i needed a way to force a re-render on the RSP side when dragSession's state changed, that way the value of isVirtualDragging would update, but there wasn't any other logic to go with it which is why it's empty

Reply:
including an empty useEffect won't force a re-render, the only things that cause a render are setState and prop changes
what was the issue you saw if you didn't include it?

Reply:
Ahh I see..so the issue was that when I called `isVirtualDraggging` by itself in RSP DropZone, the value was always false because of most of the re-rendering happens in RAC DropZone and so the value wasn't getting updated. Basically, I just needed a way to for RSP DropZone to know when to update this value.

Reply:
calling the `useDragSession` should be sufficient since that is what will subscribe you to when `startDragging` and `endDragging` are called in DragManager via the setState it sets up (the return type isn't needed, just the setState being called for a rerender)